numba/np/arrayobj.py

```diff
@@ -4331,6 +4331,28 @@ def array_ascontiguousarray(context, builder, sig, args):
     return _as_layout_array(context, builder, sig, args, output_layout='C')
 
 
+def _array_ascontiguous_scalar(a):
+    """common between ascontiguousarray and asfortranarray"""
+    if isinstance(a, (types.Number, types.Boolean, )):
+        dtype = a
+
+        def impl(a):
+            ary = np.empty((1, ), dtype=dtype)
+            ary[0] = a
+            return ary
+        return impl
+
+
+@overload(np.ascontiguousarray)
+def array_ascontiguousarray_scalar(a):
+    return _array_ascontiguous_scalar(a)
+
+
+@overload(np.asfortranarray)
+def array_asfortran_scalar(a):
```

Comment:
Personally I think it would be helpful if you could add a docstring here explaining this handles the scalar implementation and that the array cases are handled with the lower level glue_* APIs. Someone reading through this who is use to searching for overloads may be surprised when this is all they find.

Reply:
Agree, will add docstrings for it.

Reply:
Added the docstrings.